src/main/java/com/scg/stop/proposal/service/ProposalService.java

```diff
@@ -0,0 +1,157 @@
+        if (!proposal.isAuthorized(requestUser)) throw new BadRequestException(ExceptionCode.NOT_AUTHORIZED);
+        proposal.update(
+                proposalUpdateRequest.getTitle(),
+                convertProjectTypesToString(proposalUpdateRequest.getProjectTypes()),
+                proposalUpdateRequest.getEmail(),
+                proposalUpdateRequest.getWebSite(),
+                proposalUpdateRequest.getContent(),
+                proposalUpdateRequest.getIsVisible(),
+                proposalUpdateRequest.getIsAnonymous()
+        );
+//        emailService.sendEmail();
+        return ProposalDetailResponse.of(proposal.getId(), proposal.getUser().getName(), proposal.getEmail(), proposal.getWebSite(),
+                proposal.getTitle(), convertStringToProjectTypes(proposal.getProjectTypes()), proposal.getContent(), proposal.getProposalReply() != null);
+    }
+
+    @Transactional
+    public ProposalReplyResponse createProposalReply(Long proposalId,
+                                                     ProposalReplyRequest proposalReplyCreateRequest) {
+        Proposal proposal = proposalRepository.findById(proposalId)
+                .orElseThrow(() -> new BadRequestException(ExceptionCode.NOT_FOUND_PROPOSAL));
+        ProposalReply proposalReply = ProposalReply.createProposalReply(proposalReplyCreateRequest.getTitle(),
+                proposalReplyCreateRequest.getContent(),
+                proposal);
+        proposalReplyRepository.save(proposalReply);
+        return ProposalReplyResponse.of(proposalReply.getId(), proposalReply.getTitle(), proposalReply.getContent());
+    }
+
+    @Transactional
+    public ProposalReplyResponse updateProposalReply(Long proposalReplyId,
+                                                     ProposalReplyRequest proposalReplyUpdateRequest) {
+        ProposalReply proposalReply = proposalReplyRepository.findById(proposalReplyId)
+                .orElseThrow(() -> new BadRequestException(ExceptionCode.NOT_FOUND_PROPOSALREPLY));
+
+        proposalReply.update(proposalReplyUpdateRequest.getTitle(), proposalReplyUpdateRequest.getContent());
+        return ProposalReplyResponse.of(proposalReply.getId(), proposalReply.getTitle(), proposalReply.getContent());
+    }
+
+    @Transactional
+    public void deleteProposalReply(Long proposalReplyId) {
+        ProposalReply proposalReply = proposalReplyRepository.findById(proposalReplyId)
+                .orElseThrow(() -> new BadRequestException(ExceptionCode.NOT_FOUND_PROPOSALREPLY));
+
+        proposalReplyRepository.delete(proposalReply);
+    }
+
+    @Transactional
+    public void deleteProposal(Long proposalId, User requestUser) {
+        Proposal proposal = proposalRepository.findById(proposalId)
+                .orElseThrow(() -> new BadRequestException(ExceptionCode.NOT_FOUND_PROPOSAL));
+        if (proposal.isAuthorized(requestUser)) throw new BadRequestException(ExceptionCode.NOT_AUTHORIZED);
```

Comment:
여기 조건문이 거꾸로 된 것 같습니다!!!
```suggestion
        if (!proposal.isAuthorized(requestUser)) throw new BadRequestException(ExceptionCode.NOT_AUTHORIZED);
```

Reply:
[00b0beb]